lib/puppet/parser/functions/systemd_escape.rb

```diff
@@ -0,0 +1,15 @@
+require 'puppet/util/execution'
+
+module Puppet::Parser::Functions
+  newfunction(:systemd_escape, :type => :rvalue, :doc => <<-EOS
+  Returns a file system path as escaped by systemd.
+  https://www.freedesktop.org/software/systemd/man/systemd.unit.html#String%20Escaping%20for%20Inclusion%20in%20Unit%20Names
+  EOS
+  ) do |args|
+    raise Puppet::ParseError, ("validate_cmd(): wrong number of arguments (#{args.length}; must be 1)") if args.length != 1
+    path = args[0]
+    cmd = "/bin/systemd-escape --path #{path}"
```

Comment:
The function will be run on the master, not the agents.  Could this be a problem?  (Perhaps the server isn't a systemd based system and doesn't have this command available).

Reply:
Arguably, Linux OSes without systemd are getting more and more rare these days…

Reply:
Probably ok then (so long as they all put that binary under `/bin`)

Reply:
```
$ which systemd-escape
/usr/bin/systemd-escape
$ facter  os
{
  architecture => "x86_64",
  distro => {
    codename => "n/a",
    description => "Arch Linux",
    id => "Arch",
    release => {
      full => "rolling",
      major => "rolling"
    },
    specification => "1.4"
  },
  family => "Archlinux",
  hardware => "x86_64",
  name => "Archlinux",
  release => {
    full => "4.19.10-arch1-1-ARCH",
    major => "4",
    minor => "19"
  },
  selinux => {
    enabled => false
  }
}
```

Reply:
Systemd: the unique layer to standardize everything on Linux OSes :smile:

Reply:
nitpick: It isn't the fault from the systemd team:
* https://www.archlinux.org/news/binaries-move-to-usrbin-requiring-update-intervention/
* https://lists.archlinux.org/pipermail/arch-dev-public/2012-March/022625.html

Reply:
Do we actually need to provide an absolute path? I thought puppet is able to figure this out by itself?

Reply:
We could use `/usr/bin/env`?